# Movies Tonight Part 3: SQL DML

In [1]:
%load_ext sql
%sql sqlite:///Movies.db

'Connected: @Movies.db'

1. **Load the `DATASET.csv` file into the `Movies.db` database.**  
    - Import the CSV file to create a new table called `DATASET`. Follow the example [here](http://www.sqlitetutorial.net/sqlite-import-csv/).
    - Use a [`PRAGMA table-info` pseudo-query](https://www.sqlite.org/pragma.html#pragma_table_info) to inspect the design of the new table. [How are we going to handle dates](https://www.sqlite.org/datatype3.html)?

.open Movies.db <br>
.mode csv <br>
.import DATASET.csv DATASET <br>
.table <br>
.quit <br>

In [2]:
%%sql
PRAGMA table_info(MOVIES);

 * sqlite:///Movies.db
Done.


[(0, 'MID', 'INTEGER', 0, None, 1),
 (1, 'MTitle', 'TEXT', 0, None, 0),
 (2, 'Rating', 'TEXT', 0, None, 0)]

In [3]:
%%sql
select *
FROM DATASET
limit 10;

 * sqlite:///Movies.db
Done.


[('Akarakian Theatres Moreno 4 Cinemas', 'The Intersection Of Alessandro + Perris Blvds, Moreno Valley', '(909) 485-2899', 'Associate, The', '4:20 PM', 'PG-13', 'A', 'Austin Pendleton'),
 ('Akarakian Theatres Moreno 4 Cinemas', 'The Intersection Of Alessandro + Perris Blvds, Moreno Valley', '(909) 485-2899', 'Associate, The', '4:20 PM', 'PG-13', 'A', 'Bebe Neuwirth'),
 ('Akarakian Theatres Moreno 4 Cinemas', 'The Intersection Of Alessandro + Perris Blvds, Moreno Valley', '(909) 485-2899', 'Associate, The', '4:20 PM', 'PG-13', 'A', 'Dianne Wiest'),
 ('Akarakian Theatres Moreno 4 Cinemas', 'The Intersection Of Alessandro + Perris Blvds, Moreno Valley', '(909) 485-2899', 'Associate, The', '4:20 PM', 'PG-13', 'A', 'Eli Wallach'),
 ('Akarakian Theatres Moreno 4 Cinemas', 'The Intersection Of Alessandro + Perris Blvds, Moreno Valley', '(909) 485-2899', 'Associate, The', '4:20 PM', 'PG-13', 'A', 'Kenny Kerr'),
 ('Akarakian Theatres Moreno 4 Cinemas', 'The Intersection Of Alessandro + Perris Blvds, Moreno Valley', '(909) 485-2899', 'Associate, The', '4:20 PM', 'PG-13', 'A', 'Lainie Kazan'),
 ('Akarakian Theatres Moreno 4 Cinemas', 'The Intersection Of Alessandro + Perris Blvds, Moreno Valley', '(909) 485-2899', 'Associate, The', '4:20 PM', 'PG-13', 'A', 'Tim Daly'),
 ('Akarakian Theatres Moreno 4 Cinemas', 'The Intersection Of Alessandro + Perris Blvds, Moreno Valley', '(909) 485-2899', 'Associate, The', '4:20 PM', 'PG-13', 'A', 'Whoopi Goldberg'),
 ('Akarakian Theatres Moreno 4 Cinemas', 'The Intersection Of Alessandro + Perris Blvds, Moreno Valley', '(909) 485-2899', 'Associate, The', '4:20 PM', 'PG-13', 'D', 'Donald Petrie'),
 ('Akarakian Theatres Moreno 4 Cinemas', 'The Intersection Of Alessandro + Perris Blvds, Moreno Valley', '(909) 485-2899', 'Associate, The', '7:20 PM', 'PG-13', 'A', 'Austin Pendleton')]

2. **Populate the tables with data.**  

    - Use `INSERT ... SELECT ...` queries to extract data from `DATASET` for each table of your other tables. 
    - Start with the strong entities before moving on to the weak ones. Be sure to always select distinct     values. 
    - Foreign Keys get special treatment. You will have to use JOINs to pick up the PKs from the other side of the relationships. (As in Part 1, use the fact that movie names, theater names, and people names are always unique in our source data. That way you can use then for the JOIN columns. 
    - If you have to start over, then rerun your DDL queries from Part 2. 

In [15]:
%%sql
SELECT *
FROM MOVIES ; 

 * sqlite:///Movies.db
Done.


[]

In [4]:
%%sql
DELETE FROM MOVIES;
INSERT INTO MOVIES (MTitle ,Rating)
SELECT DISTINCT MTitle ,Rating
FROM DATASET;

SELECT *
FROM MOVIES
LIMIT 10; 

 * sqlite:///Movies.db
0 rows affected.
23 rows affected.
Done.


[(1, 'Associate, The', 'PG-13'),
 (2, 'Ghost & The Darkness, The', 'R'),
 (3, 'Independence Day', 'PG-13'),
 (4, 'D3: The Mighty Ducks', 'PG'),
 (5, 'Dear God', ''),
 (6, 'First Wives Club, The', 'PG-13'),
 (7, 'High School High', 'PG-13'),
 (8, 'Larger Than Life', 'PG'),
 (9, 'Mirror Has Two Faces, The', 'PG-13'),
 (10, 'Ransom', 'R')]

In [5]:
%%sql

DELETE FROM THEATERS;
INSERT INTO THEATERS (TName ,Phone , Location )
SELECT DISTINCT TName ,Phone , Location
FROM DATASET;

SELECT *
FROM THEATERS
LIMIT 5; 

 * sqlite:///Movies.db
0 rows affected.
9 rows affected.
Done.


[(1, 'Akarakian Theatres Moreno 4 Cinemas', '(909) 485-2899', 'The Intersection Of Alessandro + Perris Blvds, Moreno Valley'),
 (2, 'Cinema Star The Ultraplex 14', '(909) 342-2256', 'Mission Grove, Riverside'),
 (3, 'General Cinema Rancho 6', '(714) 370-2085', ' I-215 At Mt. Vernon S. At I-10, San Bernardino'),
 (4, 'Pacific Inland Center', '(714) 381-1611', ' Inland Center Mall, San Bernardino'),
 (5, 'SOCAL Canyon Crest Cinema', '(909) 682-6900', 'Central Avenue South Of 60 Freeway Near Ucr, Riverside')]

In [6]:


%%sql

DELETE FROM ARTISTS;
INSERT INTO ARTISTS (AName )
SELECT DISTINCT CName
FROM DATASET;

SELECT *
FROM ARTISTS
LIMIT 5; 

 * sqlite:///Movies.db
0 rows affected.
153 rows affected.
Done.


[(1, 'Austin Pendleton'),
 (2, 'Bebe Neuwirth'),
 (3, 'Dianne Wiest'),
 (4, 'Eli Wallach'),
 (5, 'Kenny Kerr')]

In [7]:
%%sql

DELETE FROM SHOWS;
INSERT INTO SHOWS (STime , MID , TID)
SELECT DISTINCT  DATASET.Showtime , MOVIES.MID , THEATERS.TID
FROM DATASET
    JOIN MOVIES ON (MOVIES.MTitle = DATASET.MTitle)
    JOIN THEATERS ON (THEATERS.TName = DATASET.TName );

SELECT * FROM SHOWS ;

 * sqlite:///Movies.db
0 rows affected.
131 rows affected.
Done.


[(1, 1, 1, '4:20 PM'),
 (2, 1, 1, '7:20 PM'),
 (3, 1, 1, '9:40 PM'),
 (4, 2, 1, '5:10 PM'),
 (5, 2, 1, '9:00 PM'),
 (6, 2, 1, '9:05 PM'),
 (7, 3, 1, '7:00 PM'),
 (8, 1, 2, '4:40 PM'),
 (9, 1, 2, '7:30 PM'),
 (10, 4, 2, '2:30 PM'),
 (11, 5, 2, '5:15 PM'),
 (12, 5, 2, '7:45 PM'),
 (13, 6, 2, '5:30 PM'),
 (14, 6, 2, '8:00 PM'),
 (15, 2, 2, '5:00 PM'),
 (16, 2, 2, '7:40 PM'),
 (17, 7, 2, '4:00 PM'),
 (18, 7, 2, '6:15 PM'),
 (19, 7, 2, '8:30 PM'),
 (20, 8, 2, '8:20 PM'),
 (21, 9, 2, '4:45 PM'),
 (22, 9, 2, '5:45 PM'),
 (23, 9, 2, '7:45 PM'),
 (24, 9, 2, '8:45 PM'),
 (25, 10, 2, '4:15 PM'),
 (26, 10, 2, '4:45 PM'),
 (27, 10, 2, '5:15 PM'),
 (28, 10, 2, '7:00 PM'),
 (29, 10, 2, '7:30 PM'),
 (30, 10, 2, '8:00 PM'),
 (31, 10, 2, '9:45 PM'),
 (32, 11, 2, '7:30 PM'),
 (33, 12, 2, '5:00 PM'),
 (34, 12, 2, '6:00 PM'),
 (35, 12, 2, '7:15 PM'),
 (36, 12, 2, '8:15 PM'),
 (37, 13, 2, '6:00 PM'),
 (38, 1, 3, '5:00 PM'),
 (39, 1, 3, '7:30 PM'),
 (40, 5, 3, '4:45 PM'),
 (41, 5, 3, '7:15 PM'),
 (42, 6, 3, '5:30 PM'),
 (43, 6, 3, '8:00 PM'),
 (44, 14, 3, '4:00 PM'),
 (45, 14, 3, '7:15 PM'),
 (46, 15, 3, '5:15 PM'),
 (47, 15, 3, '7:45 PM'),
 (48, 11, 3, '7:00 PM'),
 (49, 2, 4, '10:05 PM'),
 (50, 2, 4, '5:15 PM'),
 (51, 2, 4, '7:40 PM'),
 (52, 16, 5, '4:40 PM'),
 (53, 16, 5, '7:10 PM'),
 (54, 16, 5, '9:40 PM'),
 (55, 17, 5, '3:50 PM'),
 (56, 2, 5, '6:50 PM'),
 (57, 2, 5, '9:15 PM'),
 (58, 3, 5, '6:10 PM'),
 (59, 3, 5, '9:10 PM'),
 (60, 14, 5, '6:20 PM'),
 (61, 14, 5, '9:10 PM'),
 (62, 18, 5, '4:25 PM'),
 (63, 18, 5, '8:05 PM'),
 (64, 19, 5, '6:20 PM'),
 (65, 19, 5, '9:15 PM'),
 (66, 15, 5, '4:30 PM'),
 (67, 15, 5, '7:00 PM'),
 (68, 15, 5, '9:30 PM'),
 (69, 20, 5, '4:10 PM'),
 (70, 20, 5, '6:30 PM'),
 (71, 20, 5, '9:00 PM'),
 (72, 21, 5, '6:40 PM'),
 (73, 21, 5, '9:25 PM'),
 (74, 22, 5, '6:00 PM'),
 (75, 4, 6, '4:15 PM'),
 (76, 5, 6, '6:40 PM'),
 (77, 5, 6, '9:00 PM'),
 (78, 6, 6, '4:20 PM'),
 (79, 6, 6, '6:30 PM'),
 (80, 6, 6, '8:40 PM'),
 (81, 17, 6, '4:50 PM'),
 (82, 2, 6, '7:20 PM'),
 (83, 2, 6, '9:50 PM'),
 (84, 15, 6, '4:30 PM'),
 (85, 15, 6, '7:00 PM'),
 (86, 15, 6, '9:30 PM'),
 (87, 20, 6, '4:25 PM'),
 (88, 20, 6, '6:50 PM'),
 (89, 20, 6, '9:15 PM'),
 (90, 11, 6, '5:30 PM'),
 (91, 11, 6, '8:30 PM'),
 (92, 23, 6, '4:40 PM'),
 (93, 23, 6, '7:15 PM'),
 (94, 23, 6, '9:40 PM'),
 (95, 9, 7, '4:10 PM'),
 (96, 9, 7, '6:45 PM'),
 (97, 9, 7, '9:25 PM'),
 (98, 10, 7, '4:20 PM'),
 (99, 10, 7, '5:20 PM'),
 (100, 10, 7, '7:00 PM'),
 (101, 10, 7, '8:00 PM'),
 (102, 10, 7, '9:40 PM'),
 (103, 12, 7, '4:45 PM'),
 (104, 12, 7, '5:50 PM'),
 (105, 12, 7, '7:15 PM'),
 (106, 12, 7, '8:10 PM'),
 (107, 12, 7, '9:30 PM'),
 (108, 6, 8, '4:20 PM'),
 (109, 6, 8, '7:10 PM'),
 (110, 6, 8, '9:30 PM'),
 (111, 7, 8, '5:00 PM'),
 (112, 7, 8, '7:00 PM'),
 (113, 7, 8, '9:00 PM'),
 (114, 15, 8, '4:30 PM'),
 (115, 15, 8, '7:00 PM'),
 (116, 15, 8, '9:45 PM'),
 (117, 12, 8, '5:25 PM'),
 (118, 12, 8, '7:30 PM'),
 (119, 12, 8, '9:35 PM'),
 (120, 5, 9, '5:15 PM'),
 (121, 5, 9, '9:25 PM'),
 (122, 2, 9, '7:15 PM'),
 (123, 8, 9, '7:30 PM'),
 (124, 9, 9, '10:20 PM'),
 (125, 9, 9, '4:30 PM'),
 (126, 9, 9, '7:40 PM'),
 (127, 20, 9, '10:05 PM'),
 (128, 20, 9, '4:40 PM'),
 (129, 20, 9, '7:20 PM'),
 (130, 11, 9, '4:20 PM'),
 (131, 11, 9, '9:30 PM')]

In [3]:
%%sql
SELECT STime , MTitle , TName
FROM SHOWS
    JOIN  MOVIES USING (MID)
    JOIN  THEATERS USING (TID);


 * sqlite:///Movies.db
Done.


[('4:20 PM', 'Associate, The', 'Akarakian Theatres Moreno 4 Cinemas'),
 ('7:20 PM', 'Associate, The', 'Akarakian Theatres Moreno 4 Cinemas'),
 ('9:40 PM', 'Associate, The', 'Akarakian Theatres Moreno 4 Cinemas'),
 ('5:10 PM', 'Ghost & The Darkness, The', 'Akarakian Theatres Moreno 4 Cinemas'),
 ('9:00 PM', 'Ghost & The Darkness, The', 'Akarakian Theatres Moreno 4 Cinemas'),
 ('9:05 PM', 'Ghost & The Darkness, The', 'Akarakian Theatres Moreno 4 Cinemas'),
 ('7:00 PM', 'Independence Day', 'Akarakian Theatres Moreno 4 Cinemas'),
 ('4:40 PM', 'Associate, The', 'Cinema Star The Ultraplex 14'),
 ('7:30 PM', 'Associate, The', 'Cinema Star The Ultraplex 14'),
 ('2:30 PM', 'D3: The Mighty Ducks', 'Cinema Star The Ultraplex 14'),
 ('5:15 PM', 'Dear God', 'Cinema Star The Ultraplex 14'),
 ('7:45 PM', 'Dear God', 'Cinema Star The Ultraplex 14'),
 ('5:30 PM', 'First Wives Club, The', 'Cinema Star The Ultraplex 14'),
 ('8:00 PM', 'First Wives Club, The', 'Cinema Star The Ultraplex 14'),
 ('5:00 PM', 'Ghost & The Darkness, The', 'Cinema Star The Ultraplex 14'),
 ('7:40 PM', 'Ghost & The Darkness, The', 'Cinema Star The Ultraplex 14'),
 ('4:00 PM', 'High School High', 'Cinema Star The Ultraplex 14'),
 ('6:15 PM', 'High School High', 'Cinema Star The Ultraplex 14'),
 ('8:30 PM', 'High School High', 'Cinema Star The Ultraplex 14'),
 ('8:20 PM', 'Larger Than Life', 'Cinema Star The Ultraplex 14'),
 ('4:45 PM', 'Mirror Has Two Faces, The', 'Cinema Star The Ultraplex 14'),
 ('5:45 PM', 'Mirror Has Two Faces, The', 'Cinema Star The Ultraplex 14'),
 ('7:45 PM', 'Mirror Has Two Faces, The', 'Cinema Star The Ultraplex 14'),
 ('8:45 PM', 'Mirror Has Two Faces, The', 'Cinema Star The Ultraplex 14'),
 ('4:15 PM', 'Ransom', 'Cinema Star The Ultraplex 14'),
 ('4:45 PM', 'Ransom', 'Cinema Star The Ultraplex 14'),
 ('5:15 PM', 'Ransom', 'Cinema Star The Ultraplex 14'),
 ('7:00 PM', 'Ransom', 'Cinema Star The Ultraplex 14'),
 ('7:30 PM', 'Ransom', 'Cinema Star The Ultraplex 14'),
 ('8:00 PM', 'Ransom', 'Cinema Star The Ultraplex 14'),
 ('9:45 PM', 'Ransom', 'Cinema Star The Ultraplex 14'),
 ('7:30 PM', 'Sleepers', 'Cinema Star The Ultraplex 14'),
 ('5:00 PM', 'Space Jam', 'Cinema Star The Ultraplex 14'),
 ('6:00 PM', 'Space Jam', 'Cinema Star The Ultraplex 14'),
 ('7:15 PM', 'Space Jam', 'Cinema Star The Ultraplex 14'),
 ('8:15 PM', 'Space Jam', 'Cinema Star The Ultraplex 14'),
 ('6:00 PM', "Stephen King's Thinner", 'Cinema Star The Ultraplex 14'),
 ('5:00 PM', 'Associate, The', 'General Cinema Rancho 6'),
 ('7:30 PM', 'Associate, The', 'General Cinema Rancho 6'),
 ('4:45 PM', 'Dear God', 'General Cinema Rancho 6'),
 ('7:15 PM', 'Dear God', 'General Cinema Rancho 6'),
 ('5:30 PM', 'First Wives Club, The', 'General Cinema Rancho 6'),
 ('8:00 PM', 'First Wives Club, The', 'General Cinema Rancho 6'),
 ('4:00 PM', 'Michael Collins', 'General Cinema Rancho 6'),
 ('7:15 PM', 'Michael Collins', 'General Cinema Rancho 6'),
 ('5:15 PM', 'Romeo & Juliet', 'General Cinema Rancho 6'),
 ('7:45 PM', 'Romeo & Juliet', 'General Cinema Rancho 6'),
 ('7:00 PM', 'Sleepers', 'General Cinema Rancho 6'),
 ('10:05 PM', 'Ghost & The Darkness, The', 'Pacific Inland Center'),
 ('5:15 PM', 'Ghost & The Darkness, The', 'Pacific Inland Center'),
 ('7:40 PM', 'Ghost & The Darkness, The', 'Pacific Inland Center'),
 ('4:40 PM', 'Big Night', 'SOCAL Canyon Crest Cinema'),
 ('7:10 PM', 'Big Night', 'SOCAL Canyon Crest Cinema'),
 ('9:40 PM', 'Big Night', 'SOCAL Canyon Crest Cinema'),
 ('3:50 PM', 'Fly Away Home', 'SOCAL Canyon Crest Cinema'),
 ('6:50 PM', 'Ghost & The Darkness, The', 'SOCAL Canyon Crest Cinema'),
 ('9:15 PM', 'Ghost & The Darkness, The', 'SOCAL Canyon Crest Cinema'),
 ('6:10 PM', 'Independence Day', 'SOCAL Canyon Crest Cinema'),
 ('9:10 PM', 'Independence Day', 'SOCAL Canyon Crest Cinema'),
 ('6:20 PM', 'Michael Collins', 'SOCAL Canyon Crest Cinema'),
 ('9:10 PM', 'Michael Collins', 'SOCAL Canyon Crest Cinema'),
 ('4:25 PM', 'Nutty Professor, The', 'SOCAL Canyon Crest Cinema'),

In [9]:
%%sql
DELETE FROM CREDITS;

INSERT INTO CREDITS (CCode , AID , MID)
   SELECT DISTINCT CCode , AID , MID
    FROM DATASET
    JOIN ARTISTS ON (ARTISTS.AName  = DATASET.CName)
    JOIN MOVIES ON (MOVIES.MTitle  = DATASET.MTitle) ;
    
SELECT * FROM CREDITS; 
    

 * sqlite:///Movies.db
0 rows affected.
162 rows affected.
Done.


[(1, 1, 1, 'A'),
 (2, 2, 1, 'A'),
 (3, 3, 1, 'A'),
 (4, 4, 1, 'A'),
 (5, 5, 1, 'A'),
 (6, 6, 1, 'A'),
 (7, 7, 1, 'A'),
 (8, 8, 1, 'A'),
 (9, 9, 1, 'D'),
 (10, 10, 2, 'A'),
 (11, 11, 2, 'A'),
 (12, 12, 2, 'A'),
 (13, 13, 2, 'A'),
 (14, 14, 2, 'A'),
 (15, 15, 2, 'A'),
 (16, 16, 2, 'A'),
 (17, 17, 2, 'A'),
 (18, 18, 2, 'D'),
 (19, 19, 3, 'A'),
 (20, 20, 3, 'A'),
 (21, 21, 3, 'A'),
 (22, 22, 3, 'A'),
 (23, 23, 3, 'A'),
 (24, 24, 3, 'A'),
 (25, 25, 3, 'A'),
 (26, 26, 3, 'A'),
 (27, 27, 3, 'A'),
 (28, 28, 3, 'A'),
 (29, 29, 3, 'A'),
 (30, 30, 3, 'D'),
 (31, 31, 4, 'A'),
 (32, 32, 4, 'A'),
 (33, 33, 4, 'A'),
 (34, 34, 4, 'A'),
 (35, 35, 4, 'A'),
 (36, 35, 4, 'D'),
 (37, 36, 5, 'A'),
 (38, 37, 5, 'A'),
 (39, 38, 5, 'A'),
 (40, 39, 5, 'A'),
 (41, 40, 5, 'A'),
 (42, 41, 5, 'A'),
 (43, 42, 5, 'A'),
 (44, 43, 5, 'D'),
 (45, 44, 6, 'A'),
 (46, 45, 6, 'A'),
 (47, 46, 6, 'A'),
 (48, 47, 6, 'D'),
 (49, 48, 7, 'A'),
 (50, 49, 7, 'A'),
 (51, 50, 7, 'A'),
 (52, 51, 7, 'A'),
 (53, 52, 7, 'D'),
 (54, 53, 8, 'A'),
 (55, 54, 8, 'A'),
 (56, 55, 8, 'A'),
 (57, 56, 8, 'A'),
 (58, 57, 8, 'A'),
 (59, 58, 8, 'A'),
 (60, 59, 8, 'D'),
 (61, 60, 9, 'A'),
 (62, 61, 9, 'A'),
 (63, 62, 9, 'A'),
 (64, 63, 9, 'A'),
 (65, 60, 9, 'D'),
 (66, 64, 10, 'A'),
 (67, 65, 10, 'A'),
 (68, 66, 10, 'A'),
 (69, 67, 10, 'A'),
 (70, 68, 10, 'A'),
 (71, 69, 10, 'D'),
 (72, 70, 11, 'A'),
 (73, 71, 11, 'A'),
 (74, 72, 11, 'A'),
 (75, 73, 11, 'A'),
 (76, 74, 11, 'A'),
 (77, 75, 11, 'A'),
 (78, 76, 11, 'A'),
 (79, 77, 11, 'A'),
 (80, 78, 11, 'A'),
 (81, 79, 11, 'A'),
 (82, 80, 11, 'A'),
 (83, 81, 11, 'A'),
 (84, 82, 11, 'D'),
 (85, 53, 12, 'A'),
 (86, 83, 12, 'A'),
 (87, 84, 12, 'A'),
 (88, 85, 12, 'A'),
 (89, 86, 12, 'D'),
 (90, 87, 13, 'A'),
 (91, 88, 13, 'A'),
 (92, 89, 13, 'A'),
 (93, 90, 13, 'A'),
 (94, 91, 13, 'A'),
 (95, 92, 13, 'D'),
 (96, 93, 14, 'A'),
 (97, 94, 14, 'A'),
 (98, 95, 14, 'A'),
 (99, 96, 14, 'A'),
 (100, 97, 14, 'A'),
 (101, 98, 14, 'D'),
 (102, 99, 15, 'A'),
 (103, 100, 15, 'A'),
 (104, 101, 15, 'A'),
 (105, 102, 15, 'A'),
 (106, 103, 15, 'A'),
 (107, 104, 15, 'A'),
 (108, 105, 15, 'D'),
 (109, 106, 16, 'A'),
 (110, 107, 16, 'A'),
 (111, 108, 16, 'A'),
 (112, 77, 16, 'A'),
 (113, 109, 16, 'A'),
 (114, 110, 16, 'A'),
 (115, 106, 16, 'D'),
 (116, 109, 16, 'D'),
 (117, 111, 17, 'A'),
 (118, 112, 17, 'A'),
 (119, 113, 17, 'A'),
 (120, 114, 17, 'D'),
 (121, 115, 18, 'A'),
 (122, 116, 18, 'A'),
 (123, 117, 18, 'A'),
 (124, 118, 18, 'A'),
 (125, 119, 18, 'A'),
 (126, 120, 18, 'A'),
 (127, 121, 18, 'D'),
 (128, 122, 19, 'A'),
 (129, 123, 19, 'A'),
 (130, 124, 19, 'A'),
 (131, 125, 19, 'A'),
 (132, 126, 19, 'D'),
 (133, 127, 20, 'A'),
 (134, 117, 20, 'A'),
 (135, 128, 20, 'A'),
 (136, 129, 20, 'A'),
 (137, 130, 20, 'A'),
 (138, 28, 20, 'A'),
 (139, 131, 20, 'D'),
 (140, 132, 21, 'A'),
 (141, 100, 21, 'A'),
 (142, 133, 21, 'A'),
 (143, 134, 21, 'A'),
 (144, 135, 21, 'A'),
 (145, 136, 21, 'D'),
 (146, 137, 22, 'A'),
 (147, 138, 22, 'A'),
 (148, 139, 22, 'A'),
 (149, 140, 22, 'A'),
 (150, 141, 22, 'A'),
 (151, 142, 22, 'A'),
 (152, 143, 22, 'A'),
 (153, 144, 22, 'A'),
 (154, 145, 22, 'A'),
 (155, 146, 22, 'A'),
 (156, 147, 22, 'D'),
 (157, 148, 4, 'A'),
 (158, 149, 23, 'A'),
 (159, 150, 23, 'A'),
 (160, 151, 23, 'A'),
 (161, 152, 23, 'A'),
 (162, 153, 23, 'D')]

3. **Check your work.**  
There are 9 theaters, 23 movies, 152 artists, 131 shows, and 161 credits.

4. **Now check the source data.**  
There are actually 13 duplicate rows in the original source data! Can you spot them? Write a SQL query that counts how many times each (MTitle, TName, Showtime, CCode, and CName) appears. Which ones appear more than once?


Used this tutorial to get the answer https://www.sqlservertutorial.net/sql-server-basics/sql-server-find-duplicates




In [23]:
%%sql
SELECT MTitle, TName, Showtime, CCode, CName ,COUNT(*) occurrences
FROM DATASET
GROUP BY MTitle, TName, Showtime, CCode, CName
HAVING 
    COUNT(*) > 1;

 * sqlite:///Movies.db
Done.


[('Dear God', 'SOCAL Canyon Springs Cinema', '9:00 PM', 'A', 'Greg Kinnear', 2),
 ('Dear God', 'SOCAL Canyon Springs Cinema', '9:00 PM', 'A', 'Hector Elizondo', 2),
 ('Dear God', 'SOCAL Canyon Springs Cinema', '9:00 PM', 'A', 'Jon Seda', 2),
 ('Dear God', 'SOCAL Canyon Springs Cinema', '9:00 PM', 'A', 'Laurie Metcalf', 2),
 ('Dear God', 'SOCAL Canyon Springs Cinema', '9:00 PM', 'A', 'Maria Pitillo', 2),
 ('Dear God', 'SOCAL Canyon Springs Cinema', '9:00 PM', 'A', 'Roscoe Lee Browne', 2),
 ('Dear God', 'SOCAL Canyon Springs Cinema', '9:00 PM', 'A', 'Tim Conway', 2),
 ('Dear God', 'SOCAL Canyon Springs Cinema', '9:00 PM', 'D', 'Garry Marshall', 2),
 ('High School High', 'United Artists Riverside (Galleria) Tyler Mall', '9:00 PM', 'A', 'Jon Lovitz', 2),
 ('High School High', 'United Artists Riverside (Galleria) Tyler Mall', '9:00 PM', 'A', 'Mekhi Phifer', 2),
 ('High School High', 'United Artists Riverside (Galleria) Tyler Mall', '9:00 PM', 'A', 'Tia Carrera', 2),
 ('High School High', 'United Artists Riverside (Galleria) Tyler Mall', '9:00 PM', 'A', 'and Louise Fletcher', 2),
 ('High School High', 'United Artists Riverside (Galleria) Tyler Mall', '9:00 PM', 'D', 'Hart Bochner', 2)]

5. **Drop the `DATASET` table.**  
We don't need it anymore. 

In [ ]:
%%sql
DROP TABLE IF EXISTS DATASET

6. **Save, commit, and push your work to GitHub.**  
Use the commit summary comment "Completed Part 3".